src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackService.java

```diff
@@ -64,136 +55,65 @@ public CreateCaseCallbackService(
-                configItem.getService(),
-                newCaseId,
-                caseIdAsString
-            );
-
-            return Validation.valid(new ProcessResult(
-                ImmutableMap.of(CASE_REFERENCE, Long.toString(newCaseId))
-            ));
-        } catch (InvalidCaseDataException exception) {
-            if (BAD_REQUEST.equals(exception.getStatus())) {
-                throw exception;
-            } else {
-                return Validation.valid(new ProcessResult(
-                    exception.getResponse().warnings,
-                    exception.getResponse().errors
-                ));
-            }
-        } catch (Exception exception) {
-            log.error(
-                "Failed to create exception for service {} and exception record {}",
-                configItem.getService(),
-                caseIdAsString,
-                exception
-            );
-
-            return Validation.invalid(Array.of("Internal error. " + exception.getMessage()));
+        Validation<String, Void> event = isCreateNewCaseEvent(request.getEventId());
+        if (event.isInvalid()) {
+            return new ProcessResult(emptyList(), singletonList(event.getError()));
+        } else {
+            return hasServiceNameInCaseTypeId(request.getCaseDetails())
+                .map(serviceName -> {
+                    try {
+                        ServiceConfigItem serviceCfg = serviceConfigProvider.getConfig(serviceName);
+                        if (serviceCfg == null || serviceCfg.getTransformationUrl() == null) {
+                            return new ProcessResult(emptyList(), singletonList("Transformation URL is not configured"));
+                        } else {
+                            return validator
+                                .getValidation(request.getCaseDetails())
+                                .map(exceptionRecord -> {
+                                    SuccessfulTransformationResponse transformationResp =
+                                        transformationClient.transformExceptionRecord(
+                                            serviceCfg.getTransformationUrl(),
+                                            exceptionRecord,
+                                            s2sTokenGenerator.generate()
+                                        );
+                                    long newCaseId = createNewCaseInCcd(
+                                        idamToken,
+                                        userId,
+                                        exceptionRecord.poBoxJurisdiction,
+                                        transformationResp.caseCreationDetails,
+                                        request.getCaseDetails().getId().toString()
+                                    );
+                                    return new ProcessResult(ImmutableMap.of(CASE_REFERENCE, Long.toString(newCaseId)));
```

Comment:
This method is long and there's a lot of wrapping in it. At least this bit should be extracted out.

Reply:
won't help, you'd still have to pass many parameters and wrap.
See how it was before. with both transformation and case creation in single method.

Reply:
The new method would take 5 parameters. Not great, but it's the same number as an already existing call here. In my opinion, the method would be much more readable if this whole bit was extracted. As of now, it's hard to tell what part the following `.getOrElseGet(..` corresponds with.